test-runner/src/main.ts

```diff
@@ -33,24 +34,19 @@ async function main() {
   await policy.execute(async (context) => {
     logger.info(`Attempt ${context.attempt}...`)
 
-    const response = await fetch(url, {
-      method: 'POST',
-      mode: 'cors',
-      body: JSON.stringify(args),
+    const response = await httpClient().post<TestResponse>(url.toString(), JSON.stringify(args), {
```

Comment:
This should be `httpClient.post` not `httpClient().post`. 

Besides, this could stay as `fetch` because using the same `httpClient` with test cases has no benefit also no harm. Up to you.